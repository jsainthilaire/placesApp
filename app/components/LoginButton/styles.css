.button {
    color: #fff;
    cursor: pointer;
    border-radius: 5px;
    border-width: 0;
    font-size: 25px;
    margin: 5px;
    padding: 15px;
    width: 80%;
}

.google {
    composes: button;
    background: #dd5347;
}

.google:hover {
    background: #a92014;
}

.facebook {
    composes: button;
    background: #4267b2;
}

.facebook:hover {
    background: #365899;
    border-color: #29487d;
}
