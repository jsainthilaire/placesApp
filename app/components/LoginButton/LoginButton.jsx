import React from 'react'
import PropTypes from 'prop-types'
import get from 'lodash/get'
import { google, facebook } from './styles.css'


const providers = {
  google: {
    text: 'Login with Google',
    className: google,
  },
  facebook: {
    text: 'Login with Facebook',
    className: facebook,
  },
}

const LoginButton = ({ provider, onAuth }) => (
  <button className={get(providers, `${provider}.className`)} onClick={onAuth(provider)}>
    {get(providers, `${provider}.text`)}
  </button>
)

LoginButton.defaulProps = {
  provider: providers.google,
}

LoginButton.propTypes = {
  provider: PropTypes.string,
  onAuth: PropTypes.func.isRequired,
}

export default LoginButton
