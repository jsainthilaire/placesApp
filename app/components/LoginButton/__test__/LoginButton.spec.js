import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import noop from 'lodash/noop'
import LoginButton from '../LoginButton'

const minProps = {
  onAuth: noop,
}

test('LoginButton should render correctly', () => {
  const wrapper = shallow(<LoginButton {...minProps} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

test('Login should call onAuth if the user clicks on it', () => {
  const onAuth = jest.fn()
  const wrapper = shallow(<LoginButton {...minProps} onAuth={onAuth} />)
  wrapper.find('button').simulate('click')
  expect(onAuth).toBeCalled()
  expect(toJson(wrapper)).toMatchSnapshot()
})
