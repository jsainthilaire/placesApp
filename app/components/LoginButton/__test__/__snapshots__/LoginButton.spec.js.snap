// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should call onAuth if the user clicks on it 1`] = `<button />`;

exports[`LoginButton should render correctly 1`] = `<button />`;
